pytorch_lightning/plugins/training_type/ipu.py

```diff
@@ -0,0 +1,325 @@
+            autoreport_dir: Optional directory to store autoReport output.
+            convert_model_to_half: Converts the model to half precision, which can be used for pure FP16 training.
+        """
+        super().__init__(parallel_devices, cluster_environment)
+        if not poptorch.ipuHardwareIsAvailable():
+            raise MisconfigurationException("IPU Accelerator requires IPUs to run.")
+
+        self.convert_model_to_half = convert_model_to_half
+        self.device_iterations = device_iterations
+        self.autoround_num_ipus = autoround_num_ipus
+        self.autoreport = autoreport
+        self.autoreport_dir = autoreport_dir
+        self.poptorch_models = {}
+        self._original_accumulate_grad_batches = None
+
+        if self.autoreport:
+            options = {"autoReport.all": self.autoreport}
+            if self.autoreport_dir:
+                if not os.path.exists(self.autoreport_dir):
+                    os.makedirs(self.autoreport_dir)
+                options["autoReport.directory"] = self.autoreport_dir
+            os.environ["POPLAR_ENGINE_OPTIONS"] = json.dumps(options)
+
+    @property
+    def lightning_module(self) -> Optional[LightningModule]:
+        return self.model.module if isinstance(self.model, LightningIPUModule) else self.model
+
+    def pre_dispatch(self) -> None:
+        self._handle_gradient_accumulation_steps()
+        if self.convert_model_to_half:
+            log.info('Using full 16bit precision, converting LightningModule weights to FP16.')
+            self.model = self.model.half()
+        precision = self.lightning_module.trainer.precision
+        precision = 16 if self.convert_model_to_half else precision
+
+        model = LightningIPUModule(self.lightning_module, precision)
+        self.model = model
+
+        # Separate models are instantiated for different stages, but they share the same weights on host.
+        # When validation/test models are run, weights are synced first.
+
+        if self.lightning_module.trainer.training:
+            # Create model for training which will run training.
+            optimizer = self.lightning_module.trainer.optimizers[0]
+            model = poptorch.trainingModel(model=model, options=self._create_opts(training=True), optimizer=optimizer)
+            self.poptorch_models['train'] = model
+        for x in ('val', 'test', 'predict'):
+            model = poptorch.inferenceModel(
+                model=model,
+                options=self._create_opts(training=False),
+            )
+            self.poptorch_models[x] = model
+
+    @property
+    def replication_factor(self):
+        return len(self.parallel_devices)
+
+    def _create_opts(self, training: bool):
+        opts = poptorch.Options()
+        opts.deviceIterations(self.device_iterations)
+        opts.replicationFactor(self.replication_factor)
+        gradient_accumulation = self.lightning_module.trainer.accumulate_grad_batches if training else 1
```

Comment:
Can `accumulate_grad_batches` be a dictionary ?

Reply:
Did one better, added a test and throw an exception for now!